src/test/deploy/functions/checkIam.spec.ts

```diff
@@ -253,4 +430,174 @@ describe("checkIam", () => {
+            `serviceAccount:service-${projectNumber}@gcp-sa-pubsub.iam.gserviceaccount.com`,
+          ],
+        },
+        {
+          role: checkIam.RUN_INVOKER_ROLE,
+          members: [`serviceAccount:${projectNumber}-compute@developer.gserviceaccount.com`],
+        },
+        {
+          role: checkIam.EVENTARC_EVENT_RECEIVER_ROLE,
+          members: [`serviceAccount:${projectNumber}-compute@developer.gserviceaccount.com`],
+        },
+        {
+          role: checkIam.EVENTARC_SERVICE_AGENT_ROLE,
+          members: [
+            `serviceAccount:service-${projectNumber}@gcp-sa-eventarc.iam.gserviceaccount.com`,
+          ],
+        },
+      ],
+    };
+    getIamStub.resolves({
+      etag: "etag",
+      version: 3,
+      bindings: [BINDING],
+    });
+    setIamStub.resolves(newIamPolicy);
+    const wantFn: backend.Endpoint = {
+      id: "wantFn",
+      entryPoint: "wantFn",
+      platform: "gcfv2",
+      eventTrigger: {
+        eventType: "google.firebase.firebasealerts.alerts.v1.published",
+        eventFilters: [
+          {
+            attribute: "alerttype",
+            value: "crashlytics.newFatalIssue",
+          },
+        ],
+        retry: false,
+      },
+      ...SPEC,
+    };
+
+    await checkIam.ensureServiceAgentRoles(projectNumber, backend.of(wantFn), backend.empty());
+
+    expect(getIamStub).to.have.been.calledOnce;
+    expect(setIamStub).to.have.been.calledOnce;
+    expect(setIamStub).to.have.been.calledWith(projectNumber, newIamPolicy, "bindings");
+  });
+
+  it("should not add bindings for a new v2 alerts function with v2 deployed functions", async () => {
+    const newIamPolicy = {
+      etag: "etag",
+      version: 3,
+      bindings: [BINDING],
+    };
+    getIamStub.resolves({
+      etag: "etag",
+      version: 3,
+      bindings: [BINDING],
+    });
+    setIamStub.resolves(newIamPolicy);
+    const wantFn: backend.Endpoint = {
+      id: "wantFn",
+      entryPoint: "wantFn",
+      platform: "gcfv2",
+      eventTrigger: {
+        eventType: "google.firebase.firebasealerts.alerts.v1.published",
+        eventFilters: [
+          {
+            attribute: "alerttype",
+            value: "crashlytics.newFatalIssue",
+          },
+        ],
+        retry: false,
+      },
+      ...SPEC,
+    };
+    const haveFn: backend.Endpoint = {
+      id: "haveFn",
+      entryPoint: "haveFn",
+      platform: "gcfv2",
+      eventTrigger: {
+        eventType: "google.cloud.storage.object.v1.finalized",
+        eventFilters: [
+          {
+            attribute: "bucket",
+            value: "my-bucket",
+          },
+        ],
+        retry: false,
+      },
+      ...SPEC,
+    };
+
+    await checkIam.ensureServiceAgentRoles(projectNumber, backend.of(wantFn), backend.of(haveFn));
+
+    expect(getIamStub).to.have.been.calledOnce;
+    expect(setIamStub).to.have.been.calledOnce;
+    expect(setIamStub).to.have.been.calledWith(projectNumber, newIamPolicy, "bindings");
```

Comment:
aside - why does setIam gets called if no new binding is introduced?

Reply:
ahh good catch, added a check to return early if we don't have new bindings to add